apps/gallery/test/marionette/activity_test.js

```diff
@@ -0,0 +1,130 @@
+    prefs: {
+      'device.storage.enabled': true,
+      'device.storage.testing': true,
+      'device.storage.prompt.testing': true
+    },
+
+    apps: apps
+  });
+
+  var galleryApp, activityCallerApp;
+  var imageInfo = {
+      name: 'pictures/firefoxOS.png',
+      type: 'image/png'
+    };
+
+  setup(function() {
+    // Remove all files in temp device storage.
+    client.fileManager.removeAllFiles();
+    // Add file into the pictures directory
+    client.fileManager.add({
+      type: 'pictures',
+      filePath: 'test_media/Pictures/firefoxOS.png'
+    });
+    galleryApp = new Gallery(client);
+    activityCallerApp = GalleryActivityCaller.create(client);
+  });
+
+  test('pick image using gallery pick activity', function() {
+    activityCallerApp.launch();
+    // Initiate pick activity by clicking pick image button
+    activityCallerApp.pickImage();
+
+    // Select image from thumbnail list in gallery app
+    galleryApp.selectImage();
+
+    client.waitFor(function(){
+      return galleryApp.editCropCanvas.displayed();
+    });
+
+    galleryApp.cropDoneButton.click();
+    activityCallerApp.switchTo();
+
+    client.waitFor(function(){
+      return activityCallerApp.pickedImageName.displayed();
+    });
+
+    // Compare returned blob name and type with the
+    // image loaded in gallery app
+    var pickedImageName = activityCallerApp.pickedImageName.text();
+    assert.strictEqual(pickedImageName, imageInfo.name);
+
+    var pickedImageType = activityCallerApp.pickedImageType.text();
+    assert.strictEqual(pickedImageType, imageInfo.type);
+
+  });
+
+  test('share image using gallery share activity', function() {
+    galleryApp.launch();
+    galleryApp.tapFirstThumbnail();
+    galleryApp.shareImage('Fake Gallery Activity Caller');
+    activityCallerApp.switchTo();
+
+    client.waitFor(function(){
+      return activityCallerApp.sharedImageName.displayed();
+    });
+
+    // Compare shared blob name and type with the
+    // image loaded in gallery app
+
+    var sharedImageName = activityCallerApp.sharedImageName.text();
+    assert.strictEqual(sharedImageName, imageInfo.name);
+
+    var sharedImageType = activityCallerApp.sharedImageType.text();
+    assert.strictEqual(sharedImageType, imageInfo.type);
+  });
+
+
+  test('open image using gallery open activity', function() {
+    activityCallerApp.launch();
+    // Initiate pick activity by clicking pick image button
+    activityCallerApp.pickImage();
+
+    // Select image from thumbnail list in gallery app
+    galleryApp.selectImage();
+
+    client.waitFor(function(){
+      return galleryApp.editCropCanvas.displayed();
+    });
+
+    galleryApp.cropDoneButton.click();
+    activityCallerApp.switchTo();
+
+    client.waitFor(function(){
+      return activityCallerApp.pickedImageName.displayed();
+    });
+
+    // Tap picked image to view image using gallery open activity
+    activityCallerApp.tapPickedImage();
+    galleryApp.switchTo();
```

Comment:
This is a tricky test because you should have two different instances of the gallery app; one for the pick and one for the open. That seems like something that would be good to verify.  But maybe it would be easier if the activityCallerApp could get the image with its own device storage call instead of a pick activity. As it stands you're testing the pick activity twice here